app/src/main/java/io/homeassistant/companion/android/sensors/StorageSensorManager.kt

```diff
@@ -125,29 +125,21 @@ class StorageSensorManager : SensorManager {
         var suffix = ""
 
         var sizeLong = size
-        if (size >= 1024) {
+        if (sizeLong >= 1024) {
             suffix = "KB"
             sizeLong /= 1024
-            if (size >= 1024) {
+            if (sizeLong >= 1024) {
                 suffix = "MB"
                 sizeLong /= 1024
-                if (size >= 1024) {
+                if (sizeLong >= 1024) {
                     suffix = "GB"
                     sizeLong /= 1024
                 }
             }
         }
 
-        val resultBuffer = StringBuilder(sizeLong.toString())
-
-        var commaOffset = resultBuffer.length - 3
-        while (commaOffset > 0) {
-            resultBuffer.insert(commaOffset, ',')
-            commaOffset -= 3
-        }
-
-        resultBuffer.append(suffix)
-        return resultBuffer.toString()
+        val sizeWithThousandsSeparator = String.format("%,d", sizeLong)
+        return "$sizeWithThousandsSeparator$suffix"

```

Comment:
I just found that this method needs some improvement, but it isn't strictly related to this PR's scope.

Is it OK to have them in this PR or should I create a new PR for them?

Reply:
I think it's fine to update the scope from "fix updating the storage sensors" to "fix the storage sensors", strictly speaking refactoring is already out of scope 😉

Reply:
I edited the PR's name and description (and also reworded the commit message of 1f9e57f8 → new: 6980a2e9)